packages/contracts-bedrock/src/L1/SystemConfigInterop.sol

```diff
@@ -0,0 +1,77 @@
+///         the L2 chain.
+contract SystemConfigInterop is SystemConfig {
+    /// @notice Error thrown when the OptimismPortal address is not set.
+    error OptimismPortalNotSet();
+
+    /// @custom:semver 2.2.0+interop
+    function version() public pure override returns (string memory) {
+        return string.concat(super.version(), "+interop");
+    }
+
+    /// @notice Internal setter for the gas paying token address, includes validation.
+    ///         The token must not already be set and must be non zero and not the ether address
+    ///         to set the token address. This prevents the token address from being changed
+    ///         and makes it explicitly opt-in to use custom gas token. Additionally,
+    ///         OptimismPortal's address must be non zero, since otherwise the call to set the
+    ///         config for the gas paying token to OptimismPortal will fail.
+    /// @param _token Address of the gas paying token.
+    function _setGasPayingToken(address _token) internal override {
+        if (_token != address(0) && _token != Constants.ETHER && !isCustomGasToken()) {
+            if (optimismPortal() == address(0)) revert OptimismPortalNotSet();
+            require(
+                ERC20(_token).decimals() == GAS_PAYING_TOKEN_DECIMALS, "SystemConfig: bad decimals of gas paying token"
+            );
+            bytes32 name = GasPayingToken.sanitize(ERC20(_token).name());
+            bytes32 symbol = GasPayingToken.sanitize(ERC20(_token).symbol());
+
+            // Set the gas paying token in storage and in the OptimismPortal.
+            GasPayingToken.set({ _token: _token, _decimals: GAS_PAYING_TOKEN_DECIMALS, _name: name, _symbol: symbol });
+            OptimismPortal(payable(optimismPortal())).setConfig(
+                ConfigType.GAS_PAYING_TOKEN, abi.encode(_token, GAS_PAYING_TOKEN_DECIMALS, name, symbol)
+            );
+        }
+    }
+
+    /// @notice Adds a chain to the interop dependency set. Can only be called by the owner.
+    /// @param _chainId Chain ID of chain to add.
+    function addDependency(uint256 _chainId) external onlyOwner {
+        _addDependency(_chainId);
+    }
+
+    /// @notice Internal function for adding a chain to the interop dependency set.
+    ///         OptimismPortal must be set before calling this function.
+    /// @param _chainId Chain ID of chain to add.
+    function _addDependency(uint256 _chainId) internal {
+        if (optimismPortal() == address(0)) revert OptimismPortalNotSet();
+
+        OptimismPortal(payable(optimismPortal())).setConfig(ConfigType.ADD_DEPENDENCY, abi.encode(_chainId));
+    }
+
+    /// @notice Removes a chain from the interop dependency set. Can only be called by the owner.
+    /// @param _chainId Chain ID of the chain to remove.
+    function removeDependency(uint256 _chainId) external onlyOwner {
+        _addDependency(_chainId);
```

Comment:
I don't think we need to make internal functions, the logic is reused anywhere else

Reply:
this is more for consistency with how `SystemConfig` sets values, exposing an external method with modifier `onlyOwner` and then an internal method called within it. should we preserve consistency or opt for dropping methods like `_addDependency`?

Reply:
You are right that consistency is important. The big reason they are abstracted like that is so they can be set in the initializer without the owner needing to make the call, which we do not provide a way to add deps via the initializer at the moment. Perhaps this is something that we want to consider, but it would add extra complexity with the deploy script, so I think its fine to skip. Everything is tradeoffs, right now its consistency vs simplicity

Reply:
makes sense!